Framework/DataHandling/src/ISISDataArchive.cpp

```diff
@@ -53,8 +53,14 @@ std::string ISISDataArchive::getArchivePath(const std::set<std::string> &filenam
   }
 
   for (const auto &filename : filenames) {
-    const std::string path_without_extension = getPath(filename);
+    std::string path_without_extension = getPath(filename);
     if (!path_without_extension.empty()) {
+#ifdef __APPLE__
+      // Replace Linux path with macOS path if we're on a Mac.
+      constexpr char *unixString = "/archive/";
```

Comment:
Can we replace these c style strings with `std::strings` or `constexpr std::string_view` if you want to keep them `constexpr`?

Using `constexpr char *unixString = "string literal";` will throw warnings on some compilers, as it was superseded by `const char unixString[] = "/archive/";`, which in itself we probably want to avoid in new code.

As there's only one other instance of the use of c style strings in this file we could look at making the change there too.